prow/scripts/cluster-integration/kyma-integration-k3d-telemetry.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+
+export KYMA_SOURCES_DIR="./kyma"
+
+function prereq_test() {
+  command -v node >/dev/null 2>&1 || { echo >&2 "node not found"; exit 1; }
+  command -v npm >/dev/null 2>&1 || { echo >&2 "npm not found"; exit 1; }
+  command -v jq >/dev/null 2>&1 || { echo >&2 "jq not found"; exit 1; }
+  command -v helm >/dev/null 2>&1 || { echo >&2 "helm not found"; exit 1; }
+  command -v kubectl >/dev/null 2>&1 || { echo >&2 "kubectl not found"; exit 1; }
+  command -v k3d >/dev/null 2>&1 || { echo >&2 "k3d not found"; exit 1; }
+  command -v go >/dev/null 2>&1 || { echo >&2 "go not found"; exit 1; }
+}
+
+function load_env() {
+  ENV_FILE=".env"
+  if [ -f "${ENV_FILE}" ]; then
+    # shellcheck disable=SC2046
+    export $(xargs < "${ENV_FILE}")
+  fi
+  export PATH="${PATH}:/usr/local/go/bin"
+  export PATH="${PATH}:~/go/bin"
+}
+
+function install_cli() {
+  local install_dir
+  declare -r install_dir="/usr/local/bin"
+  mkdir -p "$install_dir"
+
+  local os
+  os="$(uname -s | tr '[:upper:]' '[:lower:]')"
+  if [[ -z "$os" || ! "$os" =~ ^(darwin|linux)$ ]]; then
+    echo >&2 -e "Unsupported host OS. Must be Linux or Mac OS X."
+    exit 1
+  else
+    readonly os
+  fi
+
+  pushd "$install_dir" || exit
+  curl -Lo kyma "https://storage.googleapis.com/kyma-cli-stable/kyma-${os}"
+  chmod +x kyma
+  popd
+
+  kyma version --client
+}
+
+function deploy_kyma() {
+  k3d version
+
+  if [[ -v K8S_VERSION ]]; then
+    echo "Creating k3d with kuberenetes version: ${K8S_VERSION}"
+    kyma provision k3d --ci -k "${K8S_VERSION}"
+  else
+    kyma provision k3d --ci
+  fi
+
+  echo "Printing client and server version info"
+
+  kubectl version
+
+  local kyma_deploy_cmd
+  kyma_deploy_cmd="kyma deploy -p evaluation --ci --source=local --workspace ${KYMA_SOURCES_DIR}"
+
+  if [[ -v TELEMETRY_TRACING_ENABLED ]]; then
+    kyma_deploy_cmd+=" --value=telemetry.operator.controllers.tracing.enabled=true"
+    ls ${KYMA_SOURCES_DIR}/components/telemetry-operator/config/crd/
+    if [[ -f ${KYMA_SOURCES_DIR}/components/telemetry-operator/config/crd/bases/telemetry.kyma-project.io_tracepipelines.yaml ]]; then
+        echo "Copy tracepipeline CRD"
+        cp ${KYMA_SOURCES_DIR}/components/telemetry-operator/config/crd/bases/telemetry.kyma-project.io_tracepipelines.yaml ${KYMA_SOURCES_DIR}/installation/resources/crds/telemetry/tracepipelines.crd.yaml
+    fi
+  fi
+
+  if [[ -v TELEMETRY_ENABLED ]]; then
+    kyma_deploy_cmd+=" --value=global.telemetry.enabled=true"
+    kyma_deploy_cmd+=" --components-file kyma-integration-k3d-telemetry-components.yaml"
+  fi
+
+  $kyma_deploy_cmd
+
```

Comment:
We also need to dry-run the deployments with prod profile after the normal deployment. It will not apply anything just render the manifests with production values and we want to know if there are errors when trying to helm template:
`kyma deploy --dry-run -p production` + other args that are set here in general